lib/adal-angular.js

```diff
@@ -60,270 +65,279 @@ if (typeof module !== 'undefined' && module.exports) {
-                    }
-                };
-            }];
+                        }
+                    };
+                    
+                    // Route change event tracking to receive fragment and also auto renew tokens
+                    $rootScope.$on('$routeChangeStart', routeChangeHandler);
+                    
+                    $rootScope.$on('$stateChangeStart', stateChangeHandler);
+                    
+                    $rootScope.$on('$locationChangeStart', locationChangeHandler);
+                    
+                    updateDataFromCache(_adal.config.clientId);
+                    $rootScope.userInfo = _oauthData;
+                    
+                    return {
+                        // public methods will be here that are accessible from Controller
+                        config: _adal.config,
+                        login: function (scopes, policy) {
+                            _adal.login(scopes, policy);
+                        },
+                        loginInProgress: function () {
+                            return _adal.loginInProgress();
+                        },
+                        logOut: function () {
+                            _adal.logOut();
+                        //call signout related method
+                        },
+                        getCachedToken: function (scopes, policy) {
+                            return _adal.getCachedToken(scopes, policy);
+                        },
+                        userInfo: _oauthData,
+                        acquireTokenSilent: function (scopes, policy) {
+                            // automated token request call
+                            var deferred = $q.defer();
+                            _adal.acquireTokenSilent(scopes, policy, function (error, tokenOut) {
+                                if (error) {
+                                    _adal._logstatus('err :' + error);
+                                    deferred.reject(error);
+                                } else {
+                                    deferred.resolve(tokenOut);
+                                }
+                            });
+                            
+                            return deferred.promise;
+                        },
+                        getUser: function () {
+                            var deferred = $q.defer();
+                            _adal.getUser(function (error, user) {
+                                if (error) {
+                                    _adal._logstatus('err :' + error);
+                                    deferred.reject(error);
+                                } else {
+                                    deferred.resolve(user);
+                                }
+                            });
+                            
+                            return deferred.promise;
+                        },
+                        getScopesForEndpoint: function (endpoint) {
```

Comment:
I don't think you need 'getScopesForEndpoint' and 'getPolicyForEndpoint' methods since adal returns the config.